src/main/java/emissary/LoggingPrintStream.java

```diff
@@ -0,0 +1,106 @@
+    /**
+     * The prefix used for "throwable" messages.
+     */
+    public static final String THROWABLE_PREFIX = "Exception on ";
+    /**
+     * The string used to separate the stream name from the message for "normal" messages.
+     */
+    public static final String NORMAL_SEPARATOR = " : ";
+    /**
+     * The log string used to identify substitutions.
+     */
+    public static final String LOG_SUBSTITUTION = "{}";
+    /**
+     * The log format used for "throwable" messages.
+     */
+    public static final String THROWABLE_LOG_FORMAT = THROWABLE_PREFIX + LOG_SUBSTITUTION;
+    /**
+     * The log format used for "normal" messages.
+     */
+    public static final String NORMAL_LOG_FORMAT = LOG_SUBSTITUTION + NORMAL_SEPARATOR + LOG_SUBSTITUTION;
+    /**
+     * The name of the stream to be added to the log message.
+     */
+    private final String streamName;
+    /**
+     * The logger to be used for logging messages.
+     */
+    private final Logger logger;
+    /**
+     * The number of lines to skip when a Throwable is received.
+     */
+    private int linesToSkip;
+
+    /**
+     * Constructs a custom PrintStream that logs anything it receives.
+     * 
+     * @param outputStream where the received data is to be written or null if it is to be discarded.
+     * @param streamName the name of the outputStream to be included in the log messages.
+     * @param logger that is to be used to log the messages.
+     */
+    public LoggingPrintStream(final OutputStream outputStream, final String streamName, final Logger logger) {
+        super(outputStream == null ? NullOutputStream.NULL_OUTPUT_STREAM : outputStream);
+
+        Validate.notNull(streamName, "Required: streamName not null");
+        Validate.notNull(logger, "Required: logger not null");
+
+        this.streamName = streamName;
+        this.logger = logger;
+    }
+
+    @Override
+    public void print(final String string) {
+        if (linesToSkip == 0) {
+            logger.info(NORMAL_LOG_FORMAT, streamName, string);
+        }
+
+        super.print(string);
+    }
+
+    @Override
+    public void println(final Object object) {
+        if (object instanceof Throwable) {
+            final Throwable throwable = (Throwable) object;
+            final StringWriter stringWriter = new StringWriter();
+            final PrintWriter printWriter = new PrintWriter(stringWriter);
+
+            throwable.printStackTrace(printWriter);
+
+            final String throwableString = stringWriter.toString();
+
+            try (StringReader stringReader = new StringReader(throwableString);
+                    LineNumberReader lineNumberReader = new LineNumberReader(stringReader)) {
+                lineNumberReader.skip(Long.MAX_VALUE);
+                linesToSkip = lineNumberReader.getLineNumber();
+            } catch (IOException e) {
+                // Should never happen because the Readers are backed by a String.
```

Comment:
Reserve your experience and judgment but should we still have a message in the catch even though it should never be hit?